src/main/java/model/ShopController.java

```diff
@@ -48,4 +51,18 @@ public String createShop(@RequestParam(value = "name") String name,
         repository.save(shop);
         return "redirect:/shops/" + shop.getId();
     }
+
+    @PostMapping("shops/{id}/newProd")
+    public String createShop(@RequestParam(value = "description") String desc, @PathVariable("id") long id) {
+        Shop shop = repository.findById(id);
+        Product prod = new Product();
```

Comment:
Why not use the product constructor to create the product being added (rather than create, then set individually)?
If you don't want to deal with the lack of images, you could just create another constructor that doesn't have that parameter.